addons/account/models/account_code_mapping.py

```diff
@@ -71,4 +72,16 @@ def _compute_code(self):
 
     def _inverse_code(self):
         for record in self:
-            record.account_id.with_company(record.company_id._origin).code = record.code
+            allowed_company_ids = (record.company_id | self.env.companies).ids
+            record.account_id.with_context({'allowed_company_ids': allowed_company_ids}).code = record.code
```

Comment:
Not sure I get this diff, `with_company` should do the same thing AFAIK

Reply:
I agree!